src/kubernetes.cc

```diff
@@ -167,6 +166,9 @@ json::value KubernetesReader::ComputePodAssociations(const json::Object* pod)
           ? top_level_controller->Get<json::String>("kind")
           : "Pod";
 
+  const json::Object* spec = pod->Get<json::Object>("spec");
+  const std::string node_name = spec->Get<json::String>("nodeName");
```

Comment:
What happens when node name is not set? Is the value null or empty?

Reply:
If node name is not set, this will throw a `json::Exception`. However, given that our query filters the pods that run on this node, those pods will definitely have a node name.
That said, we should figure out what to do about node-less (unscheduled) pods. I've added a comment.

Reply:
Great point, SGTM!